src/api/mappers/characterMapper.ts

```diff
@@ -3,73 +3,77 @@ import { CharacterNovel, Character, CharacterTrait, CharacterInstance, Character
+    id: dto.id,
+    aliasId: dto.aid,
+    visualNovelId: dto.vid,
+    note: dto.note,
+  });
 
-/**
- * Gets linked novels from array.
- * @param data Array of novels.
- */
-const visualNovelsFromArray = (data: [number, number, number, string][]): CharacterNovel[] => data.map(novel => ({
-  visualNovelId: novel[0],
-  releaseId: novel[1],
-  spoilerLevel: novel[2],
-  role: novel[3] as Roles,
-}));
+  /**
+   * Transforms array of numbers into array of objects with traits.
+   * @param data Array of arrays with number.
+   */
+  const mapTraitsFromDto = (data: CharacterDto['traits']): readonly CharacterTrait[] => data?.map(trait => ({
+    id: trait[0],
+    spoilerLevel: trait[1],
+  })) ?? [];
 
-/**
- * Maps dto into model.
- * @param dto Dto.
- */
-export const characterFromDto = (dto: CharacterDto): Character => ({
-  id: dto.id,
-  name: dto.name,
-  originalName: dto.original,
-  gender: dto.gender,
-  spoilerGender: dto.spoil_gender,
-  bloodType: dto.bloodt,
-  birthday: `${dto.birthday[0]}-${dto.birthday[1]}`,
-  aliases: dto.aliases,
-  description: dto.description,
-  age: dto.age,
-  image: dto.image,
-  imageFlagging: dto.image_flagging ? imageFlaggingFromDto(dto.image_flagging) : null,
-  bust: dto.bust,
-  waist: dto.waist,
-  hip: dto.hip,
-  height: dto.height,
-  weight: dto.weight,
-  cupSize: dto.cup_size,
-  traits: dto.traits ? traitsFromArray(dto.traits) : undefined,
-  voicedActors: dto.voiced.map(voicedDto => characterVoicedFromDto(voicedDto)),
-  instances: dto.instances?.map(instanceDto => characterInstanceFromDto(instanceDto)),
-  visualNovels: dto.vns ? visualNovelsFromArray(dto.vns) : undefined,
-});
+  /**
+   * Gets linked novels from array.
+   * @param data Array of novels.
+   */
+  const mapVisualNovelsFromDto = (data: CharacterDto['vns']): readonly CharacterNovel[] => data?.map(novel => ({
+    visualNovelId: novel[0],
+    releaseId: novel[1],
+    spoilerLevel: novel[2],
+    role: novel[3] as Roles,
+  })) ?? [];
+
+  /**
+   * Maps dto into model.
+   * @param dto Dto.
+   */
+  export const fromDto = (dto: CharacterDto): Character => ({
+    id: dto.id,
+    name: dto.name,
+    originalName: dto.original,
+    gender: dto.gender,
+    spoilerGender: dto.spoil_gender,
+    bloodType: dto.bloodt,
+    birthday: `${dto.birthday[0]}-${dto.birthday[1]}`,
+    aliases: dto.aliases,
+    description: dto.description,
+    age: dto.age,
+    image: dto.image,
+    imageFlagging: dto.image_flagging ? imageFlaggingFromDto(dto.image_flagging) : null,
+    bust: dto.bust,
+    waist: dto.waist,
+    hip: dto.hip,
+    height: dto.height,
+    weight: dto.weight,
+    cupSize: dto.cup_size,
+    traits: dto.traits ? mapTraitsFromDto(dto.traits) : undefined,
+    voicedActors: dto.voiced.map(voicedDto => mapCharacterVoicedFromDto(voicedDto)),
+    instances: dto.instances?.map(instanceDto => mapCharacterInstanceFromDto(instanceDto)),
+    visualNovels: dto.vns ? mapVisualNovelsFromDto(dto.vns) : undefined,
```

Comment:
Can we make nullish values consistent? 
Or is it intended for `imageFlagging` to be different from `traits` and `visualNovels`?

Reply:
@moetazaneta It's kinda hard to make it, since we can fetch only part of some entity. We have to decide, whether we fetch only full models and after we have to adjust our dto/domain layer, or we want to have ability to fetch parts of entities to prevent overfetching(we will face caching problem). 

If we will stick with first aproach, we have to fix all optional fields and sync it with official api docs.

Reply:
#83